internal/resources/aws/fsx_openzfs_file_system.go

```diff
@@ -0,0 +1,147 @@
+	StorageCapacityGB   int64
+	Region              string
+	DeploymentType      string
+	DataCompression     string
+	CompressionSavings  *int64   `infracost_usage:"compression_savings"`
+	BackupStorageGB     *float64 `infracost_usage:"backup_storage_gb"`
+}
+
+var FSxOpenZFSFileSystemUsageSchema = []*schema.UsageItem{
+	{Key: "backup_storage_gb", ValueType: schema.Float64, DefaultValue: 0},
+}
+
+func (r *FSxOpenZFSFileSystem) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *FSxOpenZFSFileSystem) BuildResource() *schema.Resource {
+	return &schema.Resource{
+		Name: r.Address,
+		CostComponents: []*schema.CostComponent{
+			r.throughputCapacityCostComponent(),
+			r.provisionedIOPSCapacityCostComponent(),
+			r.storageCapacityCostComponent(),
+			r.backupGBCostComponent(),
+		},
+		UsageSchema: FSxOpenZFSFileSystemUsageSchema,
+	}
+}
+
+func (r *FSxOpenZFSFileSystem) throughputCapacityCostComponent() *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "Throughput capacity",
+		Unit:            "MBps",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(r.ThroughputCapacity)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("AmazonFSx"),
+			ProductFamily: strPtr("Provisioned Throughput"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "deploymentOption", Value: strPtr("Single-AZ")},
+				{Key: "fileSystemType", Value: strPtr("OpenZFS")},
+			},
+		},
+	}
+}
+
+func (r *FSxOpenZFSFileSystem) provisionedIOPSCapacityCostComponent() *schema.CostComponent {
+	var provisionedIOPS = decimalPtr(decimal.NewFromInt(0))
+	if r.ProvisionedIOPSMode == "USER_PROVISIONED" {
+		provisionedIOPS = decimalPtr(decimal.NewFromFloat(math.Max(0, float64(r.ProvisionedIOPS-(3*r.StorageCapacityGB)))))
+	}
+	return &schema.CostComponent{
+		Name:            "Provisioned IOPS",
+		Unit:            "IOPS",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: provisionedIOPS,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("AmazonFSx"),
+			ProductFamily: strPtr("Provisioned IOPS"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "deploymentOption", Value: strPtr("Single-AZ")},
+				{Key: "fileSystemType", Value: strPtr("OpenZFS")},
+			},
+		},
+	}
+}
+
+func (r *FSxOpenZFSFileSystem) storageCapacityCostComponent() *schema.CostComponent {
+	var storageCapacity *decimal.Decimal
+	var compressionEnabled = ""
+	var compressionSavings int64
+	if r.DataCompression != "" && r.DataCompression != "NONE" {
+		if r.CompressionSavings != nil {
+			compressionSavings = *r.CompressionSavings
+		} else {
+			compressionSavings = int64(50)
+		}
+		compressionEnabled = fmt.Sprintf(" (%s compression, %d percent)", r.DataCompression, compressionSavings)
```

Comment:
Since `compressionEnabled` will be again used inside an `fmt.Sprintf()` it needs to be "double-escaped".
```suggestion
		compressionEnabled = fmt.Sprintf(" (%s compression, %d%%%%)", r.DataCompression, compressionSavings)
```
BUT fun fact: I tried that and the output of the unit and price gets shifted left one character in that line.
Looks like that
```
        --- Expected
        +++ Actual
        @@ -5,3 +5,3 @@
          ├─ Throughput capacity                                                1,024  MBps                 $266.24
        - ├─ SSD storage (LZ4 compression, 50 percent)                            150  GB                    $13.50
        + ├─ SSD storage (LZ4 compression, 50%)                                  150  GB                    $13.50
          └─ Backup storage                                             Monthly cost depends on usage: $0.05 per GB
        @@ -10,3 +10,3 @@
          ├─ Throughput capacity                                                1,024  MBps                 $266.24
        - ├─ SSD storage (LZ4 compression, 40 percent)                            180  GB                    $16.20
        + ├─ SSD storage (LZ4 compression, 40%)                                  180  GB                    $16.20
          └─ Backup storage                                             Monthly cost depends on usage: $0.05 per GB
```
So I don't know where that's coming from and therefore I went for the word.

Reply:
Interesting, it seems to be a go test output related error.
I did 707428a and let it update the golden file and the output looks correct. Don't know what happens behind the scene.

Reply:
Interesting, that's an odd one!